make.js

```diff
@@ -244,9 +235,39 @@ CLI.serverBuild = function(/** @type {{ node: string; task: string }} */ argv) {
     banner('Build successful', true);
 }
 
-function buildTask(taskName, taskListLength) {
+function getNodeVersion (taskName) {
+    var taskJsonPath = path.join(genTaskPath, taskName, "task.json");
+    // We prefer tasks in _generated folder because they might contain node20 version
+    // while the tasks in Tasks/ folder still could use only node16 handler 
+    if (fs.existsSync(taskJsonPath)) {
+        console.log(`Found task.json for ${taskName} in _generated folder`);
+    } else {
+        taskJsonPath = path.join(tasksPath, taskName, "task.json");
+        if (!fs.existsSync(taskJsonPath)) {
+            console.warn('Unable to find task.json in _generated folder and Tasks folder, defaulting to use Node 10');
+            return 10;
+        }
+        console.log(`Found task.json for ${taskName} in Tasks folder`)
+    }
+
+    var taskJsonContents = fs.readFileSync(taskJsonPath, { encoding: 'utf-8' });
+    var taskJson = JSON.parse(taskJsonContents);
+    var execution = taskJson['execution'] || taskJson['prejobexecution'];
+    var nodeVersion = 10;
+    for (var key of Object.keys(execution)) {
+        const executor = key.toLocaleLowerCase();
+        if (!executor.startsWith('node')) continue;
```

Comment:
I'd make this an error ... better to ensure correctness

Reply:
Updated